pyigm/fN/fnmodel.py

```diff
@@ -0,0 +1,456 @@
+""" Class for f(N) modeling
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+import os, imp
+from scipy import interpolate as scii
+import pdb
+
+from astropy.io import fits
+from astropy.utils.misc import isiterable
+from astropy import units as u
+from astropy import constants as const
+
+
+from pyigm import utils as pyigmu
+
+
+# Path for pyigm
+pyigm_path = imp.find_module('pyigm')[1]
+
+
+class FNModel(object):
+    """A Class for f(N,X) models
+
+    Attributes
+    ----------
+    mtype : str
+      Model type for the fN
+      'Hspline' -- Hermite monotonic spline
+      'Gamma' -- Gamma function [Following Inoue+14]
+    zmnx : tuple
+      Redshift range where this model applies (zmin,zmax)
+    pivots : array
+          log NHI values for the pivots
+    param : ndarray
+      Parameter array
```

Comment:
Done.  Admittedly, these aren't meant to be modified
by the user.

I have now added explicit parameter documentation for the __init__ command.